src/ui/features/onboarding/Success/ReferralCodeWidget/ReferralCodeWidget.tsx

```diff
@@ -0,0 +1,181 @@
+            ['--button-background-hover' as string]: '#f0f0f2',
+          }}
+          onClick={onClick}
+        >
+          <HStack gap={8}>
+            <GiftIcon />
+            <GradientText>
+              <UIText kind="body/accent">Enter Referral Code</UIText>
+            </GradientText>
+          </HStack>
+        </Button>
+        <UIText
+          kind="small/regular"
+          color="var(--always-white)"
+          style={{ opacity: 0.5 }}
+        >
+          if you have one
+        </UIText>
+      </VStack>
+    </>
+  );
+}
+
+function ReferralCodeForm({
+  onSuccess,
+}: {
+  onSuccess: (pendingReferrer: ReferrerData) => void;
+}) {
+  const [referralCode, setReferralCode] = useState('');
+
+  const { mutate: applyReferralCode, ...applyReferralCodeMutation } =
+    useMutation({
+      mutationFn: ({
+        referralCode: rawReferralCode,
+      }: {
+        referralCode: string;
+      }) => {
+        return walletPort.request('uiApplyReferralCodeToAllWallets', {
+          referralCode: rawReferralCode.trim(),
+        });
+      },
+      onSuccess,
+    });
+
+  return (
+    <>
+      <Spacer height={28} />
+      <VStack gap={16} className={styles.referralCodeForm}>
+        <UIText
+          kind="body/accent"
+          color="var(--always-white)"
+          style={{ textAlign: 'left' }}
+        >
+          Referral Code
+        </UIText>
+        <form
+          onSubmit={(event) => {
+            event.preventDefault();
+
+            const form = event.currentTarget;
+            if (!form.checkValidity()) {
+              return;
+            }
+            const formData = collectData(form, {});
+            const referralCode = formData.referralCode as string;
+
+            applyReferralCode({ referralCode });
+          }}
+        >
+          <VStack gap={4}>
+            <HStack gap={8} alignItems="center">
+              <ZStack>
+                <Input
+                  className={styles.referralCodeInput}
+                  name="referralCode"
+                  autoFocus={true}
+                  boxHeight={40}
+                  placeholder="Enter code"
+                  required={true}
+                  value={referralCode}
+                  onChange={(event) => setReferralCode(event.target.value)}
+                  disabled={applyReferralCodeMutation.isLoading}
+                />
+                {referralCode.length > 0 ? (
+                  <ClearSolidIcon
+                    style={{
+                      width: 20,
+                      height: 20,
+                      cursor: 'pointer',
+                      alignSelf: 'center',
+                      justifySelf: 'end',
+                      color: 'var(--neutral-500)',
+                      marginRight: 8,
+                    }}
+                    onClick={() => {
+                      setReferralCode('');
+                      applyReferralCodeMutation.reset();
```

Comment:
Why are we resetting here? What if the user simply erases input value, you do not reset in that case. So why should this click do it? Seems unnecessary

Reply:
It seems that I previously relied on `applyReferralCodeMutation.isSuccess` somewhere (I don't recall exactly what for), but now I'm using `onSuccess`. I'll just remove the `.reset()` call

Reply:
Ok!